setup.py

```diff
@@ -78,7 +78,7 @@
     f"jaxlib=={jax_version}",
     "tomlkit; python_version < '3.11'",
     "scipy<=1.13",
-    "numpy<=2.0",
+    "numpy<2.1,!=2.0.0",
```

Comment:
Why does Catalyst have a `numpy<2.1` requirement?

Reply:
> The requirement numpy<2.1 is to keep Catalyst in line with the NumPy dependency currently employed in PennyLane

My understanding from the offline discussion was that PennyLane requires `numpy<2.1` and that we wanted to do the same in Catalyst. Perhaps I misunderstood? Would you prefer that Catalyst supports all versions of NumPy (except for 2.0.0), even while PennyLane only supports <2.1? Catalyst depends on PennyLane so we'll already implicitly be requiring `numpy<2.1`.

Reply:
Makes sense! I've changed to the requirement to `numpy!=2.0.0`.